heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -97,60 +101,152 @@ const JEST_CONFIG_PACKAGE_FOLDER: string = path.dirname(require.resolve('jest-co
+      const combinedOptions: IJestPluginOptions = {
+        ...pluginOptions,
+        configurationPath: config.value || pluginOptions?.configurationPath,
+        debugHeftReporter: debugHeftReporter.value || pluginOptions?.debugHeftReporter,
+        detectOpenHandles: detectOpenHandles.value || pluginOptions?.detectOpenHandles,
+        disableCodeCoverage: disableCodeCoverage.value || pluginOptions?.disableCodeCoverage,
+        findRelatedTests: findRelatedTests.value || pluginOptions?.findRelatedTests,
+        maxWorkers: maxWorkers.value || pluginOptions?.maxWorkers,
+        // Default to true and always pass with no tests
+        passWithNoTests: true,
+        silent: silent.value || pluginOptions?.silent,
+        testNamePattern: testNamePattern.value || pluginOptions?.testNamePattern,
+        testPathPattern: testPathPattern.value || pluginOptions?.testPathPattern,
+        testTimeout: testTimeout.value ? parseInt(testTimeout.value, 10) : pluginOptions?.testTimeout,
+        updateSnapshots: updateSnapshots.value || pluginOptions?.updateSnapshots
+      };
+      await this._runJestAsync(taskSession, heftConfiguration, combinedOptions);
+    });
+  }
+
+  /**
+   * Write the data file used by the BuildTransformer
    */
-  public static async _setupJestAsync(
-    scopedLogger: ScopedLogger,
+  private async _setupJestAsync(
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
-    debugMode: boolean,
-    buildStageProperties: IBuildStageProperties
+    options?: IJestPluginOptions
   ): Promise<void> {
-    // Write the data file used by jest-build-transform
+    const typeScriptConfigurationJson: ITypeScriptConfigurationJson | undefined =
+      await loadTypeScriptConfigurationFileAsync(heftConfiguration, taskSession.logger.terminal);
+    const partialTsconfigFile: IPartialTsconfig | undefined = await loadPartialTsconfigFileAsync(
+      heftConfiguration,
+      taskSession.logger.terminal,
+      typeScriptConfigurationJson
+    );
+
+    // Validate, and write the Jest data file used by the BuildTransformer
     await HeftJestDataFile.saveForProjectAsync(heftConfiguration.buildFolder, {
-      emitFolderNameForTests: buildStageProperties.emitFolderNameForTests || 'lib',
-      extensionForTests: buildStageProperties.emitExtensionForTests || '.js',
-      skipTimestampCheck: !buildStageProperties.watchMode,
-      // If the property isn't defined, assume it's a not a TypeScript project since this
-      // value should be set by the Heft TypeScriptPlugin during the compile hook
-      isTypeScriptProject: !!buildStageProperties.isTypeScriptProject
+      // Use as defaults for now.
+      folderNameForTests: options?.folderNameForTests || 'lib',
+      extensionForTests:
+        options?.extensionForTests || typeScriptConfigurationJson?.emitCjsExtensionForCommonJS
+          ? '.cjs'
+          : '.js',
```

Comment:
Seems like these defaults should exist inside `HeftJestDataFile` instead of here.

Reply:
The reason I like it this way is because it allows for the heft-jest-data.json file show us what the resolved values are before execution time. Allows for easier debugging in these scenarios. What do we gain by putting them in the class instead?